File: src/main/java/io/github/dqualizer/dqlang/types/rqa/definition/resiliencetest/stimulus/FailedRequestsStimulus.java
package io.github.dqualizer.dqlang.types.rqa.definition.resiliencetest.stimulus;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;


@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
@ToString(callSuper = true)
public class FailedRequestsStimulus extends ResilienceStimulus {

    private int injectionFrequency;

    public FailedRequestsStimulus(int pauseBeforeTriggeringSeconds, int experimentDurationSeconds, int injectionFrequency) {
        this.setPauseBeforeTriggeringSeconds(pauseBeforeTriggeringSeconds);
        this.setExperimentDurationSeconds(experimentDurationSeconds);
        this.injectionFrequency = injectionFrequency;
    }

    public int getInjectionFrequency() {
        return injectionFrequency;
    }
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/definition/resilience/stimulus/FailedRequestsStimulus.kt
package io.github.dqualizer.dqlang.types.rqa.definition.resilience.stimulus

data class FailedRequestsStimulus(
  override val pauseBeforeTriggeringSeconds: Int,
  override val experimentDurationSeconds: Int,
  val injectionFrequency: Int
) : ResilienceStimulus()
